skyportal/models.py

```diff
@@ -283,6 +296,40 @@ class Photometry(Base):
     instrument = relationship('Instrument', back_populates='photometry')
     thumbnails = relationship('Thumbnail', passive_deletes=True)
 
+    @hybrid_property
+    def mag(self):
+        if self.flux > 0 and self.zp is not None:
+            return -2.5 * np.log10(self.flux) + self.zp
+        else:
+            return None
+
+    @hybrid_property
+    def e_mag(self):
+        if self.flux > 0 and self.fluxerr > 0:
+            return 2.5 / np.log(10) * self.fluxerr / self.flux
+        else:
+            return None
+
+    @mag.expression
+    def mag(cls):
+        return sa.case(
+            [
+                (sa.and_(cls.flux > 0, cls.zp != None),
```

Comment:
```suggestion
                (sa.and_(cls.flux > 0, cls.zp is not None),
```

Reply:
Peculiarity of SQLalchemy, need to write this as `!=`.

Reply:
typically I've seen `colname.isnot(None)` recommended